tmk_core/protocol/chibios/usb_main.c

```diff
@@ -975,6 +975,35 @@ void send_consumer(uint16_t data) {
 #endif
 }
 
+void send_programmable_button(uint32_t data) {
+#ifdef PROGRAMMABLE_BUTTON_ENABLE
+    osalSysLock();
+    if (usbGetDriverStateI(&USB_DRIVER) != USB_ACTIVE) {
+        osalSysUnlock();
+        return;
+    }
+
+    if (usbGetTransmitStatusI(&USB_DRIVER, SHARED_IN_EPNUM)) {
```

Comment:
I'm not sure that this is the correct behavior, as it's assuming a shared endpoint, which may not be accurate.

Reply:
The LUFA implementation of this feature also assumes that the shared endpoint is enabled. Looks like `PROGRAMMABLE_BUTTON_ENABLE` should be handled in `tmk_core/protocol.mk` in a way similar to `EXTRAKEY_ENABLE` (set `SHARED_EP_ENABLE = yes` if the feature is enabled), otherwise it's probably possible to make a broken configuration if nothing else requires the shared endpoint.

Adding something like `PROGRAMMABLE_BUTTON_SHARED_EP` would probably be overkill (maybe someone comes up with a use case that actually requires a separate endpoint; until then let's keep things simpler).